The Linux-specific Makefile is functionally decrepit. Patch it to hell!

--- Makefile.linux	2012-11-01 01:50:40.574704491 -0400
+++ Makefile.linux.new	2012-11-01 02:35:50.133446305 -0400
@@ -21,9 +21,7 @@
 
 INCDIR = ../Engine ../Common ../Common/libinclude
 LIBDIR =
-CFLAGS = -O2 -g -fsigned-char -Wfatal-errors -DALLEGRO_STATICLINK -DTHIS_IS_THE_ENGINE -DLINUX_VERSION -DDISABLE_MPEG_AUDIO -DBUILTIN_PLUGINS -DRTLD_NEXT $(shell pkg-config --cflags freetype2)
-CXXFLAGS = -fno-rtti -Wno-write-strings
-ASFLAGS =
+COMMON_FLAGS = $(addprefix -I,$(INCDIR)) -Wfatal-errors -DALLEGRO_STATICLINK -DTHIS_IS_THE_ENGINE -DLINUX_VERSION -DDISABLE_MPEG_AUDIO -DBUILTIN_PLUGINS -DRTLD_NEXT $(shell pkg-config --cflags freetype2)
 LIBS = -rdynamic $(shell allegro-config --libs) -laldmb -ldumb -Wl,-Bdynamic -ltheora -logg 
 ifeq ($(USE_TREMOR), 1)
   LIBS += -lvorbisidec
@@ -33,13 +31,13 @@
 endif
 LIBS += -lvorbisfile -lfreetype -logg -ldl -lpthread -lm -lc -lstdc++
 
-PREFIX = /usr/local
+PREFIX = $(D)/usr
 CC = gcc
 CXX = g++
-CFLAGS   := $(addprefix -I,$(INCDIR)) $(CFLAGS)
-CXXFLAGS := $(CFLAGS) $(CXXFLAGS)
-ASFLAGS  := $(CFLAGS) $(ASFLAGS)
-LDFLAGS  := -Wl,--as-needed $(addprefix -L,$(LIBDIR))
+CFLAGS   := $(CFLAGS) $(COMMON_FLAGS)
+CXXFLAGS := $(CXXFLAGS) $(COMMON_FLAGS)
+ASFLAGS  := $(ASFLAGS) $(COMMON_FLAGS)
+LDFLAGS  := $(LDFLAGS) -Wl,--as-needed $(addprefix -L,$(LIBDIR))
 
 all: ags
 
